utils/cipher.c

```diff
@@ -20,21 +20,42 @@
 status_t aes_decrypt(ta_cipher_ctx* cipher_ctx) {
   // FIXME: Add logger and some checks here
   mbedtls_aes_context ctx;
+  mbedtls_md_context_t sha_ctx;
   int status;
-  char* err;
+  char* err = NULL;
   uint8_t buf[AES_BLOCK_SIZE];
-
+  uint8_t digest[AES_BLOCK_SIZE * 2];
+  uint8_t nonce[IMSI_LEN + MAX_TIMESTAMP_LEN + 1] = {0};
   /* Create and initialise the context */
   mbedtls_aes_init(&ctx);
+  mbedtls_md_init(&sha_ctx);
+  if (mbedtls_md_setup(&sha_ctx, mbedtls_md_info_from_type(MBEDTLS_MD_SHA256), 1) != 0) {
+    err = "Failed to setup message-digest information";
```

Comment:
"Setup" is a noun and means the way in which something is arranged. "Set up" is a verb and refers to the action of putting things in order or arranging them.